tests/file_gura.rs

```diff
@@ -0,0 +1,199 @@
+#![cfg(feature = "gura")]
+
+use serde_derive::Deserialize;
+
+use config::{Config, File, FileFormat, Map, Value};
+use float_cmp::ApproxEqUlps;
+
+#[derive(Debug, Deserialize)]
+struct Place {
+    name: String,
+    longitude: f64,
+    latitude: f64,
+    favorite: bool,
+    telephone: Option<String>,
+    reviews: u64,
+    creator: Map<String, Value>,
+    rating: Option<f32>,
+}
+
+#[derive(Debug, Deserialize)]
+struct Settings {
+    debug: f64,
+    production: Option<String>,
+    place: Place,
+    #[serde(rename = "arr")]
+    elements: Vec<String>,
+}
+
+fn make() -> Config {
+    Config::builder()
+        .add_source(File::new("tests/Settings", FileFormat::Gura))
+        .build()
+        .unwrap()
+}
+
+#[test]
+fn test_file() {
+    let c = make();
+
+    // Deserialize the entire file as single struct
+    let s: Settings = c.try_deserialize().unwrap();
+
+    assert!(s.debug.approx_eq_ulps(&1.0, 2));
+    assert_eq!(s.production, Some("false".to_string()));
+    assert_eq!(s.place.name, "Torre di Pisa");
+    assert!(s.place.longitude.approx_eq_ulps(&43.722_498_5, 2));
+    assert!(s.place.latitude.approx_eq_ulps(&10.397_052_2, 2));
+    assert!(!s.place.favorite);
+    assert_eq!(s.place.reviews, 3866);
+    assert_eq!(s.place.rating, Some(4.5));
+    assert_eq!(s.place.telephone, None);
+    assert_eq!(s.elements.len(), 10);
+    assert_eq!(s.elements[3], "4".to_string());
+    if cfg!(feature = "preserve_order") {
+        assert_eq!(
+            s.place
+                .creator
+                .into_iter()
+                .collect::<Vec<(String, config::Value)>>(),
+            vec![
+                ("name".to_string(), "John Smith".into()),
+                ("username".into(), "jsmith".into()),
+                ("email".into(), "jsmith@localhost".into()),
+            ]
+        );
+    } else {
+        assert_eq!(
+            s.place.creator["name"].clone().into_string().unwrap(),
+            "John Smith".to_string()
+        );
+    }
+}
+
+#[test]
+fn test_gura_vec() {
+    let c = Config::builder()
+        .add_source(File::from_str(
+            r#"
+            hosts: [
+                "alpha",
+                "omega"
+            ]
+            "#,
+            FileFormat::Gura,
+        ))
+        .build()
+        .unwrap();
+
+    let v = c.get_array("hosts").unwrap();
+    let mut vi = v.into_iter();
+    assert_eq!(vi.next().unwrap().into_string().unwrap(), "alpha");
+    assert_eq!(vi.next().unwrap().into_string().unwrap(), "omega");
+    assert!(vi.next().is_none());
+}
```

Comment:
The test suite check is failing here with:

```
---- test_gura_vec stdout ----
thread 'test_gura_vec' panicked at /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/gura-0.5.1/src/parser.rs:1241:26:
index out of bounds: the len is 97 but the index is 97
error: test failed, to rerun pass `--test file_gura`
```

I am not sure if that's a bug in this PR or upstream. I am not sure how to troubleshoot this.

**EDIT:** I just ran the test suite with the `gura` crate reverted back to `0.3` and tests pass. Unfortunately there's no release notes / changelog for this crate, so it's unclear what happened or what the migration is 😕 

Presumably if their equivalent serde crate was used instead (_uses the `gura` crate_), this would be a non-issue 🤷‍♂️